src/Maker/MakeSchedule.php

```diff
@@ -0,0 +1,144 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Scheduler\Attribute\AsSchedule;
+use Symfony\Component\Scheduler\RecurringMessage;
+use Symfony\Component\Scheduler\Schedule;
+use Symfony\Component\Scheduler\ScheduleProviderInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+final class MakeSchedule extends AbstractMaker
+{
+    private string $scheduleName;
+    private ?string $message = null;
+
+    public function __construct(
+        private FileManager $fileManager,
+        private Finder $finder = new Finder(),
+    ) {
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:schedule';
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Create a scheduler component';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeScheduler.txt'))
+        ;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        if (!class_exists(AsSchedule::class)) {
+            // throw new \RuntimeException('You must install symfony/scheduler to create a schedule (composer require symfony/scheduler)');
+            $io->writeln('Running composer require symfony/scheduler');
+            $process = Process::fromShellCommandline('composer require symfony/scheduler');
+            $process->run();
+            $io->writeln('Scheduler successfully installed!');
+        }
+
+        // Loop over existing src/Message/* and ask which message the user would like to schedule
+        $availableMessages = ['Empty Schedule'];
+        $messageDir = $this->fileManager->getRootDirectory().'/src/Message';
+
+        if ($this->fileManager->fileExists($messageDir)) {
+            $finder = $this->finder->in($this->fileManager->getRootDirectory().'/src/Message');
+
+            foreach ($finder->files() as $file) {
+                $availableMessages[] = $file->getFilenameWithoutExtension();
+            }
+        }
+
+        $scheduleNameHint = 'MainSchedule';
+
+        // If the count is 1, no other messages were found - don't ask to create a message
+        if (1 !== \count($availableMessages)) {
+            $selectedMessage = $io->choice('Select which message', $availableMessages);
+
+            if ('Empty Schedule' !== $selectedMessage) {
+                $this->message = $selectedMessage;
+
+                // We don't want SomeMessageSchedule, so remove the "Message" suffix to give us SomeSchedule
+                $scheduleNameHint = sprintf('%sSchedule', Str::removeSuffix($selectedMessage, 'Message'));
+            }
+        }
+
+        // Otherwise call `make:message` to create a new message
```

Comment:
This could be cool todo. Idea is in the choice list we have:
0 -> Empty Schedule
1 -> New Schedule
2 -> ...

But I can't remember if a) this is actually dooable to run a maker inside a maker without duplicating code && b) if there _arn't_ any messages, or if `symfony/messenger` is _not_ installed - what should we do...

Reply:
We're going to skip this for now. Revisit at a later time...